### PR TITLE
Boost: Update copy to better describe Critical CSS feature

### LINKED ISSUE #28220: [Boost] Improvements to messaging
P2 link: pbNhbs-4Nw-p2

@augurlu has recommended a range of improvements to the copy around Critical CSS, to better explain the differences between the paid and free versions, and to explain to users when to regenerate their CSS.

See the linked P2 post for details.

### PR DESCRIPTION
**Pull request by haqadn**
Fixes #28220

## Proposed changes:

* Updated copy text in various places
* Added a ContextInfo component for rich tooltips
* Added sub paragraph to Critical CSS module
* Refactored and separated Critical CSS vs Cloud CSS status

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pbNhbs-4Nw-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:
Check if the changes requested in P2 are there and appearing correctly

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/css/components/context-info.scss [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		width: 1.1em;
+		height: 1.1em;
+		font-size: calc( ( 13 / 18 ) * 1em );
+		text-align: center;
+		border-radius: 50%;
+		border: calc( ( 2 / 18 ) * 1em ) solid $gray_30;
+		cursor: pointer;
+		line-height: 1em;
+		font-weight: 700;
+		color: $gray_30;
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/css/components/context-info.scss:21]: The figma file shows one instance of this in grey, and another in green. I've posted a question to @augurlu about it in the P2 thread here: pbNhbs-4Nw-p2#comment-12918

File: projects/plugins/boost/app/assets/src/js/elements/FoldingElement.svelte [hunk 2]
```diff
@@ -4,7 +4,7 @@
 	export let visible = false;
 
 	const toggle = () => ( visible = ! visible );
-	$: label = visible ? showLabel : hideLabel;
+	$: label = visible ? hideLabel : showLabel;
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/elements/FoldingElement.svelte:7]: 😊 Thank you for fixing this. I'm a little embarrassed it slipped through.

**Review by thingalon — COMMENTED** [review]
This is looking great. I have one minor nitpick (which isn't a blocker) - the popup on the dashboard page looks too narrow.

I think the way you've implemented a reusable component with a min-width is great - but I think that the `<ContextInfo>` in the modules page should maybe contain a div with a wider min-width to better accommodate the text?

There's one issue below I'm not sure about (the colour of the ℹ️ icon in two locations), so I've put a question into the P2 about it.

**Comment by haqadn** [comment]
Abandoning as it overlaps with #29250

**Pull request closed without merging**